---
layout: page
title: About
permalink: /about/
weight: 1
---

# **About me**
I’m a Postdoctoral Associate in [Feng Zhang’s Lab](https://www.zlab.bio) at the Broad Institute/MIT. I completed my PhD in the [Baym Lab](https://baymlab.hms.harvard.edu/) through the [Harvard Systems Biology Graduate Program](https://ssqbiophd.hms.harvard.edu), where I studied mobile genetic elements in bacteria, at the intersection of method development, microbiology, and genomics.

My work spans both experimental and computational biology, with expertise in bioinformatics, including genomics, metagenomics, RNA and protein homology, and pipeline development in HPC environments. I’m particularly interested in integrating wet lab and computational methods to better understand and engineer biological systems. Across projects, my research has involved advanced methods in data analysis, machine learning, and data visualization.


# **Publications**

## 2025
> [TIGR-Tas: A family of modular RNA-guided DNA-targeting systems in prokaryotes and their viruses](https://doi.org/10.1126/science.adv9789)
<br>Guilhem Faure\*, Makoto Saito\*, Max E Wilkinson\*, **Natalia Quinones-Olvera**, Peiyu Xu, Daniel Flam-Shepherd, Stephanie Kim, Nishith Reddy, Shiyou Zhu, Lilia Evgeniou, Eugene V Koonin, Rhiannon K Macrae, Feng Zhang
<br>*Science* (2025)

> [Efficient and robust search of microbial genomes via phylogenetic compression](https://doi.org/10.1038/s41592-025-02625-2)
<br>Karel Břinda, Leandro Lima, Simone Pignotti, **Natalia Quinones-Olvera**, Kamil Salikhov, Rayan Chikhi, Gregory Kucherov, Zamin Iqbal, Michael Baym
<br>*Nature Methods* (2025)

>[Pitfalls of bacterial pan-genome analysis approaches: a case study of *M. tuberculosis* and two less clonal bacterial species](https://doi.org/10.1093/bioinformatics/btaf219)
<br>Maximillian G Marin, **Natalia Quinones-Olvera**, Christoph Wippel, Mahboobeh Behruznia, Brendan M Jeffrey, Michael Harris, Brendon C Mann, Alex Rosenthal, Karen R Jacobson, Robin M Warren, Heng Li, Conor J Meehan, Maha R Farhat
<br>*Bioinformatics* (2025)

>[RNA-guided nucleases enable a gene drive of insertion sequences in plasmids](https://doi.org/10.1101/2025.02.20.638934)
<br>Kepler S Mears\*, Fernando W Rossine\*, **Natalia Quinones-Olvera,** Célia Souque, Michael Baym
<br>*bioRxiv* (2025)

>[Phage DisCo: targeted discovery of bacteriophages by co-culture](https://doi.org/10.1101/2024.11.22.624878)
<br>Eleanor A Rand, **Natalia Quinones-Olvera**, Kesther DC Jean, Carmen Hernandez-Perez, Sian V Owen, Michael Baym
<br>*MSystems* (2025)

> [Complete genome sequence of a virulent barcoded *Mycobacterium tuberculosis* str. Erdman commonly used for non-human primate infection studies](https://doi.org/10.1128/mra.01232-24)
<br>Maximillian G Marin, Michael R Chase, **Natalia Quinones-Olvera**, Shoko Wakabayashi, Douaa Mugahid, Sarah M Fortune, Maha R Farhat, Michael C Chao
<br>*Microbiology Resource Announcements* (2025)

## 2024
>[Diverse and abundant phages exploit conjugative plasmids](https://www.nature.com/articles/s41467-024-47416-z)
<br>**Natalia Quinones-Olvera**, Siân V Owen, Lucy M McCully, Maximillian G Marin, Eleanor A Rand, Alice C Fan, Oluremi J Martins Dosumu, Kay Paul, Cleotilde E Sanchez Castaño, Rachel Petherbridge, Jillian S Paull, Michael Baym
<br>*Nature Communications* (2024)

>(Thesis) [New Methods to Explore Mobile Genetic Elements in Bacteria](https://www.proquest.com/openview/cbbe409b20e5582be0200aed02157e11/1?cbl=18750&diss=y&pq-origsite=gscholar)
<br>**Natalia Quinones-Olvera**
<br>*Harvard University* (2024)

## 2021
>[Prophages encode phage-defense systems with cognate self-immunity](https://doi.org/10.1016/j.chom.2021.09.002)
<br>Siân V Owen, Nicolas Wenner, Charles L Dulberger, Ella V Rodwell, Arthur Bowers-Barnard, **Natalia Quinones-Olvera**, Daniel J Rigden, Eric J Rubin, Ethan C Garner, Michael Baym, Jay CD Hinton
<br>*Cell Host & Microbe* (2021)

## 2019
>[The fitness landscape of the African *Salmonella* Typhimurium ST313 strain D23580 reveals unique properties of the pBT1 plasmid](https://doi.org/10.1371/journal.ppat.1007948)
<br>Rocío Canals, Roy R Chaudhuri, Rebecca E Steiner, Siân V Owen, **Natalia Quinones-Olvera**, Melita A Gordon, Michael Baym, Michael Ibba, Jay CD Hinton
<br>*PLoS Pathogens* (2019)

## 2018
>[Non‐coding RNA analysis using the Rfam database](https://doi.org/10.1002/cpbi.51)
<br>Ioanna Kalvari, Eric P Nawrocki, Joanna Argasinska, **Natalia Quinones‐Olvera**, Robert D Finn, Alex Bateman, Anton I Petrov
<br>*Current Protocols in Bioinformatics* (2018)

>[Rfam 13.0: shifting to a genome-centric resource for non-coding RNA families](https://doi.org/10.1093/nar/gkx1038)
<br>Ioanna Kalvari, Joanna Argasinska, **Natalia Quinones-Olvera**, Eric P Nawrocki, Elena Rivas, Sean R Eddy, Alex Bateman, Robert D Finn, Anton I Petrov
<br>*Nucleic Acids Research* (2018)

## 2016
> (Thesis) [Implicaciones evolutivas de la plasticidad del RNA](http://132.248.9.195/ptd2016/julio/0747279/Index.html)
<br>**Natalia Quinones-Olvera**
<br> *Universidad Nacional Autónoma de México* (2016)

# **Training**
<div class="row">
{% include about/timeline.html %}
</div>